zerver/views/realm_filters.py

```diff
@@ -18,12 +18,19 @@ def list_filters(request: HttpRequest, user_profile: UserProfile) -> HttpRespons
 @require_realm_admin
 @has_request_variables
 def create_filter(request: HttpRequest, user_profile: UserProfile, pattern: str=REQ(),
-                  url_format_string: str=REQ()) -> HttpResponse:
+                  url_format_string: str=REQ(),
+                  stream_name: str=REQ(default='')) -> HttpResponse:
     try:
+        if stream_name != '':
+            stream, _, _ = access_stream_by_name(user_profile, stream_name)
+            stream_id = stream.id
+        else:
+            stream_id = 0
```

Comment:
`None` is generally the right choice of guard for a non-passed argument, and you can do this in REQ using `default=None`